include/odp/api/spec/queue_types.h

```diff
@@ -122,6 +122,37 @@ typedef enum odp_nonblocking_t {
 
 } odp_nonblocking_t;
 
+/**
+ * Original event order maintenance options
+ *
+ * Options to keep or ignore the original event order of a source queue. This
+ * option is relevant for (plain or parallel scheduled) queues that are
+ * destinations for events enqueued while holding an ordered queue
+ * synchronization context. By default, an ordered context maintains original
+ * event order regardless of the destination queue type. Event re-ordering may
+ * cause extra synchronization effort for implementation and a long delay before
+ * application can receive a re-ordered event from the destination queue. This
+ * is wasteful and in some cases the extra delay is not acceptable for those
+ * destination queues that do not need to maintain the original event order.
+ * Application can use ODP_QUEUE_ORDER_IGNORE option to prevent implementation
+ * from performing unnecessary event re-ordering and negative side-effects of
+ * that.
+ */
+typedef enum odp_queue_order_t {
```

Comment:
Clarification - what should happen to the events allocated by the thread? I.e.:
```c
odp_event_t e1 = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
odp_event_t e2 = odp_buffer_to_event(odp_buffer_alloc(some_bufpool);

odp_queue_enq(q1, e2);
odp_queue_enq(q1, e1);
```
With above in mind, what is the required order of `e2`? What changes if `q1` is a `ODP_QUEUE_ORDER_IGNORE` queue?

Reply:
Nothing changes. e2 will be always before e1 as those are enqueued from the same thread and ordering context. Re-ordering apply between ordering contexts (threads).

Reply:
OK, my proper code example would be:

```c
odp_event_t e1 = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
odp_event_t e2 = odp_buffer_to_event(odp_buffer_alloc(some_bufpool);

odp_queue_enq(q1, e2);
odp_queue_enq(q2, e1);
```

And of course there are 2 threads which perform this action. My question is should the ordering be performed at enq to q2 or at enq to q1? Given that it's the thread that helds the ordering context, not an event.

Reply:
OK. That makes sense.